src/client/requestManager.ts

```diff
@@ -110,6 +118,9 @@ export default class RequestManager {
     if (timer.unref) {
       timer.unref()
     }
+    if (this.promisesAwaitingResponse.has(newId)) {
```

Comment:
Do we care about the case where someone passes a number as a string? Ex. Id = '1' and a pending id = 1 (number), should we count those as the same id? (From a quick test online, map.has() seems to not treat '1' = 1)

Reply:
We'd want those to be treated differently, because they're different objects - and in the current implementation they would be, assuming rippled treats them differently (@natenichols do you know if this is true?)